src/utils/application.data.ts

```diff
@@ -50,27 +50,31 @@ export const removeFromApplicationData = (req: Request, key: string, id: string)
   const session = req.session;
   const appData: ApplicationData = getApplicationData(session);
 
-  const index = getIndexInApplicationData(req, appData, key, id);
+  const index = getIndexInApplicationData(req, appData, key, id, true);
   if (index === -1) {
     throw createAndLogErrorRequest(req, `application.data removeFromApplicationData - unable to find object in session data for key ${key} and ID ${id}`);
   }
   appData[key].splice(index, 1);
   setExtraData(session, appData);
 };
 
-export const getFromApplicationData = (req: Request, key: string, id: string): any => {
+// gets data from ApplicationData. throwable boolean indicates whether an error should be thrown if no data found.
+export const getFromApplicationData = (req: Request, key: string, id: string, throwable: boolean = true): any => {
   const appData: ApplicationData = getApplicationData(req.session);
 
-  const index = getIndexInApplicationData(req, appData, key, id);
-  if (index === -1) {
+  const index = getIndexInApplicationData(req, appData, key, id, throwable);
```

Comment:
I think a better name for this would be something like 'errorIfNotFound'. Would make the if statement on line 66 read better. Up to you though.